### ISSUE TITLE
Different Render Speed

### ISSUE THREAD

**Issue by pegasus1982** [eKoopmans/html2pdf#190]
Now I develop a barcode generator app.
Here I used this library
I used settimeout for bulk save.
But when I input 13 barcodes and 358 barcodes, the rendering speed is different.

> 13 barcodes : average 280ms
**console.log result**
`265ms html2canvas : Canvas renderer initialized(234x147 at 575.5,0) with scale 30`
> 358 barcodes : average 6500ms
**console.log result**
`6121ms html2canvas : Canvas renderer initialized(234x147 at 575.5,0) with scale 30`

I used settimeout in code,And I faced this issue.
I couldn't know why the rendering time is different in these cases.

**Comment by eKoopmans** [comment]
Hi @pegasus1982, the more you try to render the longer it will take. Each HTML element is individually processed and drawn onto a canvas by html2canvas.

Your numbers show an average of 22ms/barcode (for 13) and 18ms/barcode (for 358), so that sounds about right. If you have ideas for how the processing time could be reduced, please raise them over on the issues tracker for [html2canvas](https://github.com/niklasvh/html2canvas).

Closing, as this is related to html2canvas (and likely can't be improved).

**Comment by pegasus1982** [comment]
Hi @eKoopmans , I meant rendering speed is for individual barcode.
So that

> 13 barcodes : 280ms/barcode , total time = 3600ms
> 358 barcodes : 600ms/barcode, total time = 2340,000ms = 2340s
>

**Comment by eKoopmans** [comment]
Hi @pegasus1982, I see - that's definitely a big difference. I recommend testing with just html2canvas on its own (no html2pdf) and see if the time difference still exists. If so, make an issue over there, if not we'll try to follow up on it here. Thanks!

**Comment by pegasus1982** [comment]
I tried with just html2canvas.
But same issue

**Comment by eKoopmans** [comment]
Okay, if it's an issue directly in html2canvas there's not much I can do. Have you opened an issue over there?

If you do, try to give him an example where he can reproduce the problem, and keep it simple (without involving html2pdf). I'm not sure there's anything he can do to speed it up though. Maybe you could try breaking up the process yourself, rendering one barcode at a time and adding them into the PDF separately?

**Comment by eKoopmans** [comment]
Yes, html2canvas always renders the entire page, though you can control that behaviour with the `ignoreElements` option (see [documentation](http://html2canvas.hertzen.com/configuration)), which uses a function to decide which elements to ignore.

html2pdf currently has a step before that so it should only render the element(s) you specify.
